apps/vaccitron/src/app/report-list-page/report-list-page.component.ts

```diff
@@ -72,8 +86,38 @@ export class ReportListPageComponent implements OnInit, OnDestroy {
     this.#onDestroy$.next();
   }
 
-  reportsListActionEvent($event) {
+  reportsListActionEvent($event: ReportsListAction) {
     if (!this.isLoggedIn) return this.openLoginDialog();
+    switch ($event.type) {
+      case 'comingFeedback':
+        const dialogRef = this.dialog.open(MatSpinner, {
+          direction: 'rtl',
+          autoFocus: false,
+        });
+
+        this.vaccinesReportsService
+          .updateImComing(environment.apiUrl + environment.comingFeedback, {
+            reportId: $event.payload.reportId,
+          })
+          .pipe(
+            retry(3),
+            catchError((error: HttpErrorResponse) => {
+              dialogRef.close();
+              this.dialog.open(ErrorDialog, {
```

Comment:
In case the status code is that an identical feedback already recorded, we should notify this to the user.

Reply:
What do you mean? What's that status code?

Reply:
409. I mean that if the user clicked twice on same report, he should be aware that the second time will not be recorded.